src/Microsoft.AspNet.Authentication.OAuth/OAuthAuthenticationOptions.cs

```diff
@@ -109,5 +101,12 @@ public string Caption
         /// Gets or sets the type used to secure data handled by the middleware.
         /// </summary>
         public ISecureDataFormat<AuthenticationProperties> StateDataFormat { get; set; }
+
+        /// <summary>
+        /// Defines whether access and refresh tokens should be stored in the <see cref="ClaimsPrincipal"/>
+        /// after a successful authentication. You can set this property to <see cref="false"/> to reduce
+        /// the size of the final authentication cookie.
+        /// </summary>
+        public bool SaveTokensAsClaims { get; set; } = true;
```

Comment:
Actually, would it make sense to set it to true in the base options and false in the derived options? Otherwise the default behavior of the base class is an empty identity, no?